python/ray/serve/backend_state.py

```diff
@@ -126,32 +124,24 @@ def actor_resources(self) -> Dict[str, float]:
 
     def graceful_stop(self) -> None:
         """Request the actor to exit gracefully."""
-        # NOTE: the replicas may already be stopped if we failed
-        # after stopping them but before writing a checkpoint.
-        if self._stopped:
-            return
-
         try:
             handle = ray.get_actor(self._actor_name)
             self._drain_obj_ref = handle.drain_pending_queries.remote()
         except ValueError:
-            self._stopped = True
+            pass
 
     def check_stopped(self) -> bool:
         """Check if the actor has exited."""
-        if self._stopped:
-            return True
-
         try:
             handle = ray.get_actor(self._actor_name)
             ready, _ = ray.wait([self._drain_obj_ref], timeout=0)
```

Comment:
What if _drain_obj_ref failed with exception ? based on what behavior of ray.wait() it will still fill up ready list with an element but we can only ensure its success via calling ray.get() ?

For this PR it's not changed however so not blocking.